### PR TITLE
feat: Resize enormous illustrations - EXO-60323

### PR DESCRIPTION
**Pull request by hakermi**
Prior to this change, image attachments are displayed and fetched with their original size which affects the perf on stream page and leading to a long load and render time. This PR should make sure to resize these images by using the jcr thumbnail service to create a custom thumbnails.

### CODE REVIEW

**Review by ahamdi — APPROVED** [review]

File: core/services/src/main/java/org/exoplatform/services/cms/impl/ImageUtils.java [hunk 1]
```diff
@@ -73,9 +73,15 @@ public static InputStream scaleImage(BufferedImage image, int maxWidth, int maxH
...
-        thumbImage = Scalr.resize(image, Scalr.Method.BALANCED, maxWidth, maxHeight);      
+        int originWidth = image.getWidth();
+        int originHeight = image.getHeight();
+        Scalr.Mode fitMode = Scalr.Mode.AUTOMATIC;
+        if (originWidth > originHeight) {
+          fitMode = Scalr.Mode.FIT_TO_HEIGHT;
+        } else if (originHeight > originWidth) {
+          fitMode = Scalr.Mode.FIT_TO_WIDTH;
+        }
+        thumbImage = Scalr.resize(image, Scalr.Method.QUALITY, fitMode, maxWidth, maxHeight, Scalr.OP_ANTIALIAS);
```
Comment (rdenarie) [inline core/services/src/main/java/org/exoplatform/services/cms/impl/ImageUtils.java]: Should we replace ImageUtils call by a call to ImageResizeService ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: core/services/src/main/java/org/exoplatform/services/cms/impl/ImageUtils.java (hunk 1 above)
Comment (hakermi) [reply core/services/src/main/java/org/exoplatform/services/cms/impl/ImageUtils.java]: Yes we can

Comment (rdenarie) [reply core/services/src/main/java/org/exoplatform/services/cms/impl/ImageUtils.java]: We should do it in the same task so

Comment (hakermi) [reply core/services/src/main/java/org/exoplatform/services/cms/impl/ImageUtils.java]: done

Comment (rdenarie) [reply core/services/src/main/java/org/exoplatform/services/cms/impl/ImageUtils.java]: Can you rebase the PR ?

Comment (hakermi) [reply core/services/src/main/java/org/exoplatform/services/cms/impl/ImageUtils.java]: rebased

**Review by rdenarie — APPROVED** [review]

**Pull request merged**